### ISSUE TITLE
x/playground: cannot restore code selected and deleted by pressing tab

### ISSUE THREAD

**Issue by ProximaB** [golang/go#41037]
### What version of Go are you using (`go version`)?
The playground uses the latest stable release of Go.
The current version is go1.14.7.

### Web browser and operating system
The latest Chrome, Windows 10

### What operating system and processor architecture are you using (`go env`)?
That running on https://play.golang.org/.

### How to reproduce a bug?
Select any part of text and click Tab key. Code disappear, cannot undo changes.

### What did you expect to do?
Got my code back.

### What did you see instead?
Nothing 😅

**Comment by ALTree** [comment]
Interesting. If you select some code and delete it by pressing `backspace`, `canc`, or any letter/number, you can get it back by pressing `Ctrl`+`z`, but if you delete it by pressing `Tab`, `Ctrl`+`z` doesn't work.

**Comment by ProximaB** [comment]
@ALTree I found the cause of that issue. Playground have bug in custom tab handle that erase selected code and replace it with "\t" programmatically in way that doesn't support native undo support.
Please help me to make pull request, I really would like to contribute into this project.
```js
    //playground.js
    function insertTabs(n) {
      // find the selection start and end
      var start = code[0].selectionStart;
      var end   = code[0].selectionEnd;
      // split the textarea content into two, and insert n tabs
      var v = code[0].value;
      var u = v.substr(0, start);
      for (var i=0; i<n; i++) {
        u += "\t";
      }
      u += v.substr(end);
      // set revised content
      code[0].value = u; <-- this is cause of the bug
      // reset caret position after inserted tabs
      code[0].selectionStart = start+n;
      code[0].selectionEnd = start+n;
    }
```
I create solutions for this bug that seems work on Chrome and Firefox.
Simply replace line `code[0].value =u` with this snippet of code:
```js
field = code[0]
let document = field.ownerDocument;

if (!document.execCommand('insertText', false, u)) {
    //firefox only
	field.setRangeText(
		u,
		start || 0,
		end || 0,
		'end'
	);

	field.dispatchEvent(new InputEvent('input', {
		data: u,
		inputType: 'insertText',
		isComposing: false
	}));
}
```

**Comment by ALTree** [comment]
Feel free to send a PR. `playground.js` Should be the one here: https://github.com/golang/website/tree/master/content/static.

**Comment by ProximaB** [comment]
@ALTree I made pull-request https://github.com/golang/website/pull/13
Thanks :)

**Comment by distinctdan** [comment]
Just ran into this and lost some work, super obnoxious bug. Any progress on fixing this?

**Comment by ProximaB** [comment]
> Just ran into this and lost some work, super obnoxious bug. Any progress on fixing this?

**Use this [goplay.tools](goplay.tools)**

I fixed that issue, unfortunately it hasn't been merged because I've used execCommand which is described as obsolete in Mozilla documentation. I didn't find any other solutions to fix that on others browsers Blink/Webkit based without reimplementing code editor. The case here is to invoke native edit event to maintain textbook capabilities like undo, redo etc.
